src/canvas/heatmapFeature.js

```diff
@@ -96,27 +96,58 @@ var canvas_heatmapFeature = function (arg) {
    */
   ////////////////////////////////////////////////////////////////////////////
   this._createCircle = function () {
-    var circle, ctx, r, r2, blur;
+    var circle, ctx, r, r2, blur, gaussian;
     r = m_this.style('radius');
     blur = m_this.style('blurRadius');
-    if (!m_this._circle || m_this._circle.radius !== r ||
-        m_this._circle.blurRadius !== blur) {
+    gaussian = m_this.style('gaussian');
+    if (!m_this._circle || m_this._circle.gaussian !== gaussian ||
+        m_this._circle.radius !== r || m_this._circle.blurRadius !== blur) {
       circle = m_this._circle = document.createElement('canvas');
       ctx = circle.getContext('2d');
-
       r2 = blur + r;
-
       circle.width = circle.height = r2 * 2;
-      ctx.shadowOffsetX = ctx.shadowOffsetY = r2 * 2;
-      ctx.shadowBlur = blur;
-      ctx.shadowColor = 'black';
-
-      ctx.beginPath();
-      ctx.arc(-r2, -r2, r, 0, Math.PI * 2, true);
-      ctx.closePath();
-      ctx.fill();
+      if (!gaussian) {
+        ctx.shadowOffsetX = ctx.shadowOffsetY = r2 * 2;
+        ctx.shadowBlur = blur;
+        ctx.shadowColor = 'black';
+        ctx.beginPath();
+        ctx.arc(-r2, -r2, r, 0, Math.PI * 2, true);
+        ctx.closePath();
+        ctx.fill();
+      } else {
+        /* This approximates a gaussian distribution by using a 10-step
+         * piecewise linear radial gradient.  Strictly, it should not stop at
+         * the radius, but should be attenuated further.  The scale has been
+         * selected such that the values at the radius are around 1/256th of
+         * the maximum, and therefore would not be visible using an 8-bit alpha
+         * channel for the summation.  The values for opacity were generated by
+         * the python expression:
+         *   from scipy.stats import norm
+         *   for r in [0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1]:
+         *     opacity = norm.pdf(r, scale=0.3) / norm.pdf(0, scale=0.3)
```

Comment:
@aashish24, so that it has a sensible radius and fall-off.  By using a scale of 0.3, the intensity at radius=1 is approximately 1/256th of the central intensity.  This means that all of the pixels in the rendered circle have some impact on the display.  If we use a a larger number (such as the default), the circle will still have significant magnitude at the edges of our rendered area.  If we use a smaller number, then we won't use the whole area of our circle.

Simplistically, instead of using a scale, we could have scaled the radius.  The mathematics are equivalent (we could have used `norm.pdf(r/0.3) / norm.pdf(0)`).